src/t_string.c

```diff
@@ -102,6 +104,15 @@ void setGenericCommand(client *c,
 
     found = (lookupKeyWrite(c->db, key) != NULL);
 
+    /* Handle the IFEQ conditional check */
+    if ((flags & OBJ_SET_IFEQ) && found) {
+        robj *current_value = lookupKeyRead(c->db, key);
+        if (current_value == NULL || compareStringObjects(current_value, comparison) != 0) {
```

Comment:
What happens if the value currently stored isn't a STRING data type? I also think we should return an error message in that case.

Reply:
Add a test case as well.

Reply:
We return an error now!